proc/proc_windows.go

```diff
@@ -0,0 +1,369 @@
+package proc
+
+// #include "windows.h"
+// #include <stdlib.h>
+import "C"
+import (
+	"debug/pe"
+	"debug/gosym"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	sys "golang.org/x/sys/windows"
+	"path/filepath"
+	"os/exec"
+	"syscall"
+	"unsafe"
+	"runtime"
+	
+	"github.com/derekparker/delve/dwarf/line"
+	"github.com/derekparker/delve/dwarf/frame"
+)
+
+// Windows specific information.
+type OSProcessDetails struct {
+	hProcess	syscall.Handle
+	breakThread int
+}
+
+// Create and begin debugging a new process.
+func Launch(cmd []string) (*Process, error) {
+	argv0Go, err := filepath.Abs(cmd[0])
+	if err != nil {
+		return nil, err
+	}
+	// Make sure the binary exists.
+	if filepath.Base(cmd[0]) == cmd[0] {
+		if _, err := exec.LookPath(cmd[0]); err != nil {
+			return nil, err
+		}
+	}
+	if _, err := os.Stat(argv0Go); err != nil {
+		return nil, err
+	}
+
+	argv0, _ := syscall.UTF16PtrFromString(argv0Go)
+
+	pi := new(sys.ProcessInformation)
+	si := new(sys.StartupInfo)
+	err = sys.CreateProcess(argv0, nil, nil, nil, true, 2, nil, nil, si, pi)
+
+	if err != nil {
+		return nil, err
+	}
+	defer sys.CloseHandle(sys.Handle(pi.Thread))
+	
+	dbp := New(int(pi.ProcessId))
+
+	switch runtime.GOARCH {
+	case "amd64":
+		dbp.arch = AMD64Arch()
+	}
+	
+	dbp.execPtraceFunc(func() {
+		// TODO - We're ignoring the results because we assume we'll immediately hit
+		// the default breakpoint that Windows sets at process creation.
+		// Should perhaps be testing that we're not overlooking an exit event or similar?
+		_, _, err = dbp.waitForDebugEvent()
```

Comment:
I'm not certain it's even possible, but I've left the TODO here and added a note about returning `ProcessExitedError` if it is possible.